src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionfurtherevidence/HandleEvidenceEventHandler.java

```diff
@@ -136,6 +155,47 @@ private SscsDocument buildSscsDocument(SscsCaseData sscsCaseData, ScannedDocumen
             .build()).build();
     }
 
+    private DocumentLink addFooter(SscsCaseData sscsCaseData, DocumentLink url) {
+
+        String originalSenderCode = sscsCaseData.getOriginalSender().getValue().getCode();
+        String documentType = APPELLANT.getCode().equals(originalSenderCode) ? "Appellant evidence" : "Representative evidence";
+        String bundleAddition = getNextBundleAddition(sscsCaseData.getSscsDocument());
+
+        byte[] oldContent = toBytes(url.getDocumentUrl());
+        PdfWatermarker alter = new PdfWatermarker();
+        byte[] newContent;
+        try {
+            newContent = alter.shrinkAndWatermarkPdf(oldContent, String.format("%s | Appendix  %s", documentType, bundleAddition));
+        } catch (Exception e) {
+            log.error("Caught exception :" + e.getMessage(), e);
+            throw new RuntimeException(e.getMessage(), e);
+        }
+
+        ByteArrayMultipartFile file = ByteArrayMultipartFile.builder()
+                .content(newContent)
+                .name(url.getDocumentFilename())
+                .contentType(APPLICATION_PDF).build();
+
+        UploadResponse uploadResponse = evidenceManagementService.upload(singletonList(file), DM_STORE_USER_ID);
+        String location = uploadResponse.getEmbedded().getDocuments().get(0).links.self.href;
+
+        return url.toBuilder().documentUrl(location).documentBinaryUrl(location).build();
+    }
+
+    protected String getNextBundleAddition(List<SscsDocument> sscsDocument) {
+        if (sscsDocument == null) {
+            sscsDocument = new ArrayList<>();
+        }
+        String[] appendixArray = sscsDocument.stream().filter(s -> StringUtils.isNotEmpty(s.getValue().getAppendix())).map(s -> s.getValue().getAppendix()).toArray(String[]::new);
+        Arrays.sort(appendixArray);
+        if (appendixArray.length >  0) {
+            char nextChar =  (char) (StringUtils.upperCase(appendixArray[appendixArray.length - 1]).charAt(0) + 1);
```

Comment:
Have you tested what happens when it gets to Z? Believe it should go to Z1 but can't see how this would happen with this code.

Reply:
Not tested. Apparently it will not happen. Should it go to Z1?

Reply:
But it ***could*** happen right? Don't think the case should blow up if this edge case happens. Yeh Z1 seems sensible